packages/cli/src/helpers/isValidSchemaDefOverrides.ts

```diff
@@ -0,0 +1,113 @@
+import { ColumnDefProperties, SchemaDefOverrides } from '@synthql/queries';
+
+const columnDefProperties: ColumnDefProperties = {
+    type: { type: '', id: '', description: '' },
+    selectable: { type: 'boolean', const: false },
+    includable: { type: 'boolean', const: false },
+    whereable: { type: 'boolean', const: false },
+    nullable: { type: 'boolean', const: false },
+    isPrimaryKey: { type: 'boolean', const: false },
+};
+
+const columnDefPropertyKeys = Object.keys(columnDefProperties);
+
+export function isValidSchemaDefOverrides(
+    value: unknown,
+): value is SchemaDefOverrides | undefined {
+    if (isObject(value)) {
+        for (const [tableDefName, tableDefOverride] of Object.entries(value)) {
+            if (containsSingleDot(tableDefName)) {
```

Comment:
Jim why didn't you go with the JSON schema approach?

https://discord.com/channels/1222945509266817055/1222945509740642317/1263412303445889044

Reply:
Also I feel like you haven't really "grokked" the idea of returning early

Reply:
Here is a sample schema btw:

```json
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "type": "object",
    "properties": {
        "tables": {
            "type": "array",
            "items": {
                "type": "string"
            }
        },
        "schemas": {
            "type": "array",
            "items": {
                "type": "string"
            }
        },
        "defaultSchema": {
            "type": "string",
        },
        "schemaDefOverrides": {....}
    },
    "required": ["tables", "schemaDefOverrides"]
}
```